src/Nethermind/Nethermind.Network/P2P/Utils/BackgroundTaskSchedulerWrapper.cs

```diff
@@ -38,14 +38,14 @@ internal void ScheduleBackgroundTask<TReq>(TReq request, Func<TReq, Cancellation
     private async ValueTask BackgroundSyncSender<TReq, TRes>(
         (TReq Request, Func<TReq, CancellationToken, Task<TRes>> FullfillFunc) input, CancellationToken cancellationToken) where TRes : P2PMessage
     {
-        TRes response = await input.FullfillFunc.Invoke(input.Request, cancellationToken);
+        using TRes response = await input.FullfillFunc.Invoke(input.Request, cancellationToken);
         handler.Send(response);
```

Comment:
`ProtocolHandlerBase.Send` forwards response to the `Session.DeliverMessage` which eventually disposes it (inside `_packetSender.Enqueue`). Don't see any conditions under which this doesn't happen (apart from `Session` being closed when message is received).